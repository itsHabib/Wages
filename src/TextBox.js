import React, { Component } from 'react'

export default class TextBox extends Component {
    render() {
        return (
            <label for ="number-input">
                <input id="number-input"/>
            </label>
        )
    }
}