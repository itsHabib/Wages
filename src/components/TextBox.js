import React from 'react'
import TextField from 'material-ui/TextField'
import Paper from 'material-ui/Paper'


// Input field that notifies the app of numbers entered
const TextBox = ({ onNumberInput, switched }) => {

    const getNumber = (event) => {
        console.log(event.target.value)
        if (!(event.target.value === 'undefined')){
            onNumberInput(event.target.value)
        }
    }

    return (               
            <TextField
                hintText='Hourly or Annual Rate'
                type='number'
                min = '0'
                onChange={getNumber}
                id='number-input'
                    />

    )
}

module.exports = TextBox