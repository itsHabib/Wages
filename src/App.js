import React, { Component } from 'react'
import propTypes from 'prop-types'
import TextBox from './TextBox'

export default class App extends Component {
    render() {
        return (
            <div>
                <h1>Wages</h1>
                <TextBox/>
            </div>
        )
    }
}