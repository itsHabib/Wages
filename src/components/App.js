import React, { Component } from 'react'
import propTypes from 'prop-types'
import wageCalc from '../wageCalc'

import TextBox from './TextBox'
import Wages from './Wages'

export default class App extends Component { 
    constructor() {
        super()

        this.state = {
            wage: "",
            isAnnual: false
        }
        
        this.onNumberInput = this.onNumberInput.bind(this)
    }
    // Converts number to correct wage and
    // sends to wages component
    onNumberInput(number) {
        console.log(`Change detected! ${number} entered`)
        this.setState({
            wage: number
        })

    }

    render() {
        return (
            <div>
                <h1>Wages</h1>
                <TextBox onNumberInput = {this.onNumberInput}/>
                <Wages wage = {this.state.wage}/>
            </div>
        )
    }
}