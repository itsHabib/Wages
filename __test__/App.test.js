import React from 'react'
import App from '../src/App'
import TextBox from '../src/TextBox'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('<App/> - Main Container of Application', () => {

    // Initial State implies app is just loaded and
    // there is no input in the input box
    describe('Initial State of App', () => {

        // First ensure app renders correctly
        it('renders <App/> correctly', () => {
            const tree = renderer.create( < App / > ).toJSON()
            expect(tree).toMatchSnapshot()
        })

        // App is in charge of rendering the TextBox component
        it('it should render a TextBox component', () => {
            const wrapper = shallow(<App/>)
            expect(wrapper.find(TextBox)).toHaveLength(1)
        })
    })

})