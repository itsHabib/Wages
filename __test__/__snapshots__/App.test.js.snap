// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> - Main Container of Application Initial State of App renders <App/> correctly 1`] = `
<div>
  <h1>
    Wages
  </h1>
  <label
    for="number-input"
  >
    <input
      id="number-input"
    />
  </label>
</div>
`;
